src/shared/api/base.ts

```diff
@@ -1,48 +1,61 @@
 import { API_URL } from "~shared/config";
-import axios from "axios";
+import axios, { AxiosInstance } from "axios";
 import { Api } from "~shared/api/Api.ts";
 
 const ApiInstance = axios.create({
   baseURL: API_URL,
+  withCredentials: true,
   headers: {
     "Content-Type": "application/json",
   },
-  withCredentials: true
 });
+const ApiPrivateInstance = axios.create({
+  baseURL: API_URL,
+  withCredentials: true,
+  headers: {
+    "Content-Type": "application/json",
+  },
+});
+
+export function UseApi(): AxiosInstance {
```

Comment:
this solution is not clear to me
it is not necessary to create a separate instance each time to call the API